### PR TITLE
lambda-promtail: Add kinesis data stream to use in terraform

### PR DESCRIPTION
**Pull request by changhyuni**
**What this PR does / why we need it**:
https://github.com/grafana/loki/pull/5977
With the addition of the kinesis data stream function add kinesis data stream to use in terraform

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the `CONTRIBUTING.md` guide
- [ ] Documentation added
- [ ] Tests updated
- [x] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

**Comment by kavirajk** [comment]
@changhyuni can you rebase to fix the conflicts? I can merge after that.

File: tools/lambda-promtail/README.md [hunk 1]
```diff
@@ -45,14 +45,22 @@ This is the [Loki Write API](https://grafana.com/docs/loki/latest/api/#post-loki
 
 The `lambda-promtail` code picks this value up via an environment variable.
 
-Also, if your deployment requires a [VPC configuration](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#vpc_config), make sure to edit the `vpc_config` field in `main.tf` manually. Additonal documentation for the Lambda specific Terraform configuration is [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#vpc_config).
+Also, if your deployment requires a [VPC configuration](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#vpc_config), make sure to edit the `vpc_config` field in `main.tf` manually. Additonal documentation for the Lambda specific Terraform configuration is [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#vpc_config). and, If you want to link kinesis data stream to Lambda as event source, see [here](https://docs.aws.amazon.com/ko_kr/lambda/latest/dg/with-kinesis.html).
```
Comment (MichelHollands) [inline tools/lambda-promtail/README.md:48]:
```suggestion
Also, if your deployment requires a [VPC configuration](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#vpc_config), make sure to edit the `vpc_config` field in `main.tf` manually. Additonal documentation for the Lambda specific Terraform configuration is [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lambda_function#vpc_config). If you want to link kinesis data stream to Lambda as event source, see [here](https://docs.aws.amazon.com/ko_kr/lambda/latest/dg/with-kinesis.html). 
```

**Review by MichelHollands — APPROVED** [review]
LGTM with one nit

**Comment by MichelHollands** [comment]
@changhyuni Thanks for this contribution. Please rebase and then I'll merge it.

**Comment by MichelHollands** [comment]
@changhyuni You'll also need to sign the contributor license agreement. We cannot accept merge requests without it.

**Comment by changhyuni** [comment]
@MichelHollands

**Pull request closed without merging**